usr/sbin/p11sak/p11sak.c

```diff
@@ -1117,6 +1150,151 @@ static CK_ATTRIBUTE_TYPE col2type(int col)
     }
 }
 
+/**
+ *  Print in p11sak_defined_attrs.conf defined attributes in long format
+ */
+static CK_RV print_vendor(CK_SESSION_HANDLE session, CK_OBJECT_HANDLE hkey, int long_print) 
+{
+    CK_RV rc = CKR_OK;
+    int f;
+    struct ConfigBaseNode *c, *name, *hex_string, *type;
+    struct ConfigStructNode *structnode;
+    
+    if (cfg != NULL)
+    {
+        confignode_foreach(c, cfg, f) {
+            if (confignode_hastype(c, CT_STRUCT) && strcmp(c->key, "attribute") == 0) {
+                // parse...
+                structnode = confignode_to_struct(c);
+                name = confignode_find(structnode->value, "name");
+                hex_string = confignode_find(structnode->value, "id");
+                type = confignode_find(structnode->value, "type");
+
+                // checking syntax of attribute...
+                if (name == NULL) {
+                    fprintf(stderr, "Missing name in Attribute in line %hu\n", name->line);
```

Comment:
Here you cannot use `name` since it is `NULL`.  So you have to use `c` to specify the line that starts the invalid "attribute" structure.

Reply:
Done